### ISSUE TITLE
[citrix_adc]: SSLVPN pipeline Grok pattern failure for USER

### ISSUE THREAD

**Issue by untergeek** [elastic/integrations#11312]
### Integration Name

Citrix ADC [citrix_adc]

### Dataset Name

citrix_adc.log

### Integration Version

1.8.1

### Agent Version

8.15.1

### Agent Output Type

elasticsearch

### Elasticsearch Version

8.15.1

### OS Version and Architecture

ECK on x86_64

### Error Message

No message, but tracked down the problem to the `logs-citrix_adc.log-1.8.1-sslvpn_and_aaatm_feature` pipeline, the 5th pattern in the grok pipeline: 

```
^User %{USER:citrix_adc.log.user} - Client_ip %{IP:citrix_adc.log.client_ip} - Nat_ip %{IP:citrix_adc.log.nat.ip} - Vserver %{IP:citrix_adc.log.vserver.ip}:%{INT:citrix_adc.log.vserver.port} - Source %{IP:citrix_adc.log.source.ip}:%{INT:citrix_adc.log.source.port} - Destination %{IP:citrix_adc.log.destination.ip}:%{INT:citrix_adc.log.destination.port} - Start_time \"%{DATA:citrix_adc.log.start_time}\" - End_time \"%{GREEDYDATA:citrix_adc.log.end_time}\" - Duration %{DATA:citrix_adc.log.duration} - Total_bytes_send %{NUMBER:citrix_adc.log.total_bytes_send:int} - Total_bytes_recv %{NUMBER:citrix_adc.log.total_bytes_received:int} - Total_compressedbytes_send %{NUMBER:citrix_adc.log.total_compressed_bytes_send:int} - Total_compressedbytes_recv %{NUMBER:citrix_adc.log.total_compressed_bytes_recieved:int} - Compression_ratio_send %{NUMBER:citrix_adc.log.compression_ratio_send:float}% - Compression_ratio_recv %{NUMBER:citrix_adc.log.compression_ratio_recieved:float}% - Access %{WORD:citrix_adc.log.access} - Group\(s\) \"%{DATA:citrix_adc.log.groups}\"$
```

### Event Original

```
{
      "citrix": {
        "extended": {
          "message": "User redacted03@redacted.tld - Client_ip 10.0.0.1 - Nat_ip 10.0.0.1 - Vserver 172.16.0.1:443 - Source 10.0.0.1:49264 - Destination 192.168.106.5:80 - Start_time \"2024/09/30:09:23:05 \" - End_time \"2024/09/30:09:23:06 \" - Duration 00:00:01  - Total_bytes_send 1180 - Total_bytes_recv 1272 - Total_compressedbytes_send 0 - Total_compressedbytes_recv 0 - Compression_ratio_send 0.00% - Compression_ratio_recv 0.00% - Access Allowed - Group(s) \"N/A\""
        }
      }
    }
```

(Sensitive bits sanitized)

### What did you do?

I sample-tested this log against the `logs-citrix_adc.log-1.8.1-sslvpn_and_aaatm_feature` pipeline.

### What did you see?

The email-based username was not accepted by the grok pattern.

### What did you expect to see?

The email-based username should have been accepted.

### Anything else?

I tested this document and pattern in the Grok Simulator in the Dev Console.  It turns out that the very first pattern, 
`%{USER:citrix_adc.log.user}`, does not match the user in the log line: `redacted03@redacted.tld` (sanitized). 

If I changed it to `%{DATA:citrix_adc.log.user}`, it worked fine. It was then that I looked up the [grok patterns](https://github.com/logstash-plugins/logstash-patterns-core/blob/main/patterns/ecs-v1/grok-patterns):

```
USERNAME [a-zA-Z0-9._-]+
USER %{USERNAME}
EMAILLOCALPART [a-zA-Z0-9!#$%&'*+\-/=?^_`{|}~]{1,64}(?:\.[a-zA-Z0-9!#$%&'*+\-/=?^_`{|}~]{1,62}){0,63}
EMAILADDRESS %{EMAILLOCALPART}@%{HOSTNAME}
```

`USER` is an alias of `USERNAME`, and `USERNAME` will not recognize the `@` character as valid, so the grok pattern fails.

Possible solutions include just using `DATA` here, or making the grok more complicated than it needs to be to accept either `USER` or `EMAILADDRESS` as valid patterns.

**Comment by untergeek** [comment]
While someone's in there, one of the field names has a lovely typo in the same grok pattern:

`%{NUMBER:citrix_adc.log.total_compressed_bytes_recieved:int}`

Where `recieved` should read `received`

**Comment by mjwolf** [comment]
Integration citrix_adc v1.10.0 has been released and adds support for email-based user names.

Unfortunately fixing the misspelled field name would be a breaking change that will not be done at this time.  Any alerts, queries, etc. that people have set up on the current field name would break if it's changed.  I've created another issue so this problem can be tracked separately https://github.com/elastic/integrations/issues/11719
